### PR TITLE
Add gstreamer-gl-sys

### PR DESCRIPTION
**Pull request by ceyusa**
fixes issue #13 

I didn't committed the changes generated by fix.sh, should I?

### CODE REVIEW

**Comment by sdroege** [comment]
Thanks!

> I didn't committed the changes generated by fix.sh, should I?

Yes please, as a separate commit :)

**Comment by ceyusa** [comment]
> > I didn't committed the changes generated by fix.sh, should I?
> 
> Yes please, as a separate commit :)

done!

**Comment by sdroege** [comment]
Oh no, I need to build a new set of gstreamer binaries that includes libgstgl before merging this :) Later!

**Comment by sdroege** [comment]
please rebase on top of https://github.com/sdroege/gstreamer-sys/commit/549ad086d564c5a7ed5198c236a2b3a4ca882ae2

File: gstreamer-gl-sys/Cargo.toml [hunk 1]
```diff
@@ -5,42 +5,45 @@ pkg-config = "0.3.7"
...
-name = "gst-gl-sys"
+description = "FFI bindings to libgstgl-1.0"
+homepage = "https://gstreamer.freedesktop.org"
+keywords = ["ffi", "gstreamer", "gnome", "multimedia"]
+license = "MIT"
+links = "gstgl-1.0"
+name = "gstreamer-gl-sys"
+readme = "README.md"
+repository = "https://github.com/sdroege/gstreamer-sys"
 version = "0.2.0"
```
Comment (sdroege) [inline gstreamer-gl-sys/Cargo.toml]: Same version as for the other crates

File: gir-files/fix.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+xmlstarlet ed --pf --inplace \
+	   --delete '//_:constructor[@c:identifier="gst_gl_display_wayland_new_with_display"]' \
+	   --delete '//_:constructor[@c:identifier="gst_gl_display_x11_new_with_display"]' \
+	   --delete '//_:constructor[@c:identifier="gst_gl_display_egl_new_with_egl_display"]' \
+	   --delete '//_:function[@c:identifier="gst_gl_display_egl_get_from_native"]' \
+	   GstGL-1.0.gir
+
+# rename Display* to gpointer
+xmlstarlet ed --pf --inplace \
+	   --insert '//_:type[@c:type="Display*"]' -t attr -n 'name' --value 'gpointer' \
```
Comment (sdroege) [inline gir-files/fix.sh]: This is X11 stuff and shouldn't be there at all. I'd suggest removing all X11/Wayland/EGL/etc specific API from the .gir file

Comment (ceyusa) [reply gir-files/fix.sh]: Does that mean remove all the subclasses of GstDisplay?

Comment (sdroege) [reply gir-files/fix.sh]: Yes, and behind feature flags to be added back

File: gstreamer-gl-sys/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+[dev-dependencies]
+shell-words = "0.1.0"
+tempdir = "0.3"
+
+[features]
+v1_4 = []
+v1_6 = ["v1_4"]
+v1_8 = ["v1_6"]
+v1_10 = ["v1_8"]
+v1_12 = ["v1_10"]
```
Comment (sdroege) [inline gstreamer-gl-sys/Cargo.toml]: This should only really have 1.14, previous versions were API incompatible and in -bad. See webrtc for an example

File: gstreamer-gl-sys/build.rs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    } else if cfg!(feature = "v1_10") {
+        "1.10"
+    } else if cfg!(feature = "v1_8") {
+        "1.8"
+    } else if cfg!(feature = "v1_6") {
+        "1.6"
+    } else {
+        "1.0"
+    };
+
```
Comment (sdroege) [inline gstreamer-gl-sys/build.rs]: Only 1.14 here too

**Review by sdroege — COMMENTED** [review]
Almost good to go, thanks :)

**Pull request merged**